esx_service/vmdk_ops.py

```diff
@@ -290,51 +289,47 @@ def listVMDK(path):
              u'Attributes': {}} for x in vmdks]
 
 
-# Find VM , reconnect if needed. throws on error
-def findVmByName(vm_name):
-    vm = None
-    vm, _ = find_child(vm_name)
-    if not vm:
-        # try again
-        connectLocal()
-        vm, e = find_child(vm_name)
-
-    if not vm:
-        logging.error("VM {0} not found".format(vm_name))
-        raise e
-
-    return vm
-
-def find_child(vm_name):
-    e = None
+# Return VM managed object, reconnect if needed. Throws is fails twice.
+def findVmByUuid(vm_uuid):
     try:
-        vm = FindChild(GetVmFolder(), vm_name)
+        vm = si.content.searchIndex.FindByUuid(None, vm_uuid, True, False)
+        if vm:
+            return vm
     except Exception as e:
-        logging.exception("Failed to find vm: %s", vm_name) 
-        vm = None
-    return vm, e
+        logging.exception("Failed to find VM uuid=%s (traceback below), " \
+                          "retrying...", vm_uuid)
+    #
+    # Retry. It can throw in connect/search fails. But search can't return None
+    # since we get UUID from VMM so VM must exist
+    #
+    connectLocal()
+    vm = si.content.searchIndex.FindByUuid(None, vm_uuid, True, False)
+    logging.info("Found VM name=%s, id=%s ", vm.config.name, vm_uuid)
+    return vm
 
-#returns error, or None for OK
-def attachVMDK(vmdk_path, vm_name):
-    vm = findVmByName(vm_name)
+
+# Return error, or None for OK.
+def attachVMDK(vmdk_path, vm_uuid):
+    vm = findVmByUuid(vm_uuid)
     logging.info("*** attachVMDK: %s to %s VM uuid = %s",
-                 vmdk_path, vm_name, vm.config.uuid)
+                 vmdk_path, vm.config.name, vm.config.uuid)
```

Comment:
good catch - they are the same. and vm.config.uiid is just reused from the old code. Will use vm_uuid